README.md

```diff
@@ -51,8 +51,17 @@ services:
 
 By default, the IOC will be built with statically linked EPICS libraries. If
 you **need** to link them dynamically, you must define the build target as
-`dynamic-link`. This will increase the resulting image size, since unused
-dependencies will also be copied.
+`dynamic-link`.
+
+For non-static builds, an automatic pruning step is executed to make the IOC
+image size small by removing unused artifacts. This step preserves all ELF
+executables inside `RUNDIR` and the IOC repository, as well as their dependent
+EPICS modules based on the linkage information. Additional paths can be
+provided in `args` under `APP_DIRS` key to extend the list of where important
+applications and shared libraries are, including any `dlopen(3)`ed libraries.
+Pruning step can be skipped entirely by providing the `SKIP_PRUNE=1` argument,
+but this is highly discouraged as it will increase significantly the image
+size.
```

Comment:
Do you think it is interesting to specify we also prune directories from modules in use?

I don't like the idea of replicating what types of files are kept, but it may be okay to say they are also pruned.

Reply:
I think it should be mentioned that some directories (e.g. GUI ones) are assumed to be unused and removed as well.

Reply:
Done. Feel free to propose an enhanced wording ;)